lib/sshkit/scp/command.ex

```diff
@@ -3,22 +3,29 @@ defmodule SSHKit.SCP.Command do
 
   import SSHKit.Utils
 
-  def build(mode, path, options) do
-    scp = case mode do
-      :upload -> 'scp -t'
-      :download -> 'scp -f'
-    end
+  @flags [verbose: '-v', preserve: '-p', recursive: '-r']
 
-    flags = [verbose: '-v', preserve: '-p', recursive: '-r']
+  def build(:upload, path, options) do
+    scp('-t', path, options)
```

Comment:
These switches are undocumented as far as I know and not part of the "public" scp tool, that you use from the terminal. They're used internally only (so you won't find them in the `scp` man pages).

When you run `scp local.file me@remote:/tmp/remote.file` in your terminal, it'll open an SSH connection, execute `scp -t` on the remote.

- `-t` basically stands for "to" (you're copying to the remote, aka. "sink mode")
- `-f` stands for "from" (copying from the remote, aka. "source mode")

See also the link to the excellent post on the Oracle Blog above. Also `net-scp` (Ruby) has an **amazing** documentation on everything around SCP:

- https://blogs.oracle.com/janp/entry/how_the_scp_protocol_works
- https://github.com/net-ssh/net-scp/blob/6e73089f26d4ce1e5c46b4b13ea87a615fd71b69/lib/net/scp.rb#L81-L95